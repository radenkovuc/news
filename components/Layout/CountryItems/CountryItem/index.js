import styled from 'styled-components'
import React from 'react'
import {useTranslation} from 'next-i18next'

const Item = styled.div`
  display: flex;
  border-left: 1px solid #444;
  justify-content: center;
  background: ${(props) => (props.isDisabled ? '#FF6347' : props.isActive ? '#777777' : '#ffffff')};
  color: ${(props) => (props.isActive ? '#ffffff' : '#000000')};
  cursor: ${(props) => (props.isDisabled ? 'default' : 'pointer')};
  flex: auto;

  @media only screen and (min-width: 600px) {
    flex: none;
  }
`
const Text = styled.span`
  padding: 10px;
  font-size: 18px;
  font-family: 'Nunito Sans', sans-serif;

  @media only screen and (min-width: 480px) {
    padding: 20px;
    font-size: 20px;
  }
`

type Props = {
  text: String,
  isActive: Boolean,
  isDisabled: Boolean,
  onClick: Boolean
}

const CountryItem = (props: Props) => {
  const {text, isActive, isDisabled, onClick} = props
  const {t} = useTranslation()

  return (
    <Item isActive={isActive} isDisabled={isDisabled} onClick={onClick}>
      <Text>{t(text)}</Text>
    </Item>
  )
}

CountryItem.defaultProps = {
  text: '',
  expanded: false,
  isDisabled: false,
  onClick: () => {}
}

export default CountryItem
