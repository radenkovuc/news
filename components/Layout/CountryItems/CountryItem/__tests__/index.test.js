import React from 'react'
import renderer from 'react-test-renderer'

import {createTextComponent} from '../../../../../test/TestUtil'

import CountryItem from '../'

describe('<CountryItem />', () => {
  it('match snapshot default', () => {
    const tree = renderer.create(createTextComponent(<CountryItem />)).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('match snapshot with data', () => {
    const tree = renderer
      .create(createTextComponent(<CountryItem text="Text" isActive={true} isDisabled={true} />))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
