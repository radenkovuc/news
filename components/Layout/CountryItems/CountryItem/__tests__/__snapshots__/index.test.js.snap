// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryItem /> match snapshot 1`] = `
<div
  className="Item-fwh4iu-0 XAluN"
  onClick={[Function]}
>
  <span
    className="Text-fwh4iu-1 gPvOtr"
  >
    
  </span>
</div>
`;
